cohortextractor/tpp_backend.py

```diff
@@ -31,33 +31,46 @@ def __init__(self, database_url, covariate_definitions, temporary_database=None)
         self.queries = self.get_queries(self.covariate_definitions)
 
     def to_csv(self, filename):
-        queries = self.queries
+        queries = list(self.queries)
         # If we have a temporary database available we write results to a table
         # there, download them, and then delete the table. This allows us to
         # resume in the case of a failed download without rerunning the whole
         # query
         if self.temporary_database:
-            queries, cleanup_queries = self.save_results_to_temporary_db(queries)
+            output_table = self.save_results_to_temporary_db(queries)
         else:
-            cleanup_queries = []
+            output_table = "#final_output"
+            queries[-1] = (
+                f"-- Writing results into {output_table}\n"
+                f"SELECT * INTO {output_table} FROM ({queries[-1]}) t"
+            )
+            queries.append(f"CREATE INDEX ix_patient_id ON {output_table} (patient_id)")
```

Comment:
I wondered what difference, if any, making this `UNIQUE INDEX` would make. The answer is, [I think](https://explainextended.com/2009/04/23/making-an-index-unique/) "exactly none" in this case. Thought it was an interesting little article though!

Reply:
One difference it would make is that, in the case that we accidentally generate duplicate patient rows, we'd blow up when trying to create the index rather than when the unique check runs (which is after we've downloaded the results). There might be an argument for doing it that way, but I don't want to make that kind of change right now.